### ISSUE TITLE
Possible bug in Row implementation of TableSlices

### ISSUE THREAD

**Issue by lwhite1** [jtablesaw/tablesaw#878]
This needs more research, but it seems the following code works when TableSlice is converted to a table, but fails if the Rows are provided directly by the slice. It appears that the row returned by the slice is actually that of the underlying table, so that rows not part of the slice are accessed. The code is from the melt() method on Table

 ```
    for (TableSlice slice : slices) {
      // TODO: This is inefficient, but it looks like there is a bug in row iteration on slices
      Table temp = slice.asTable();

      for (Row row : temp) {
        for (String colName : measureColumnNames) {
          if (!dropMissing || !row.isMissing(colName)) {
            writeIdVariables(idVariables, result, row);
            result.stringColumn(MELT_VARIABLE_COLUMN_NAME).append(colName);
            result.doubleColumn(MELT_VALUE_COLUMN_NAME).append(row.getNumber(colName));
          }
        }
      }


```

**Comment by lwhite1** [comment]
The bug is in Row, not TableSlice. Fix to come shortly

**Comment by lwhite1** [comment]
fixed in #880
